frontend/storage/utils/apiRequestManager.ts

```diff
@@ -0,0 +1,33 @@
+import axios from 'axios';
+import { SERVER_ADRESS } from '../common';
+
+export async function apiRequestGet(endpoint: string, queryParams: { [key: string]: any }) {
+	const endpointFullString = createApiRequestUrl(endpoint, queryParams);
+	try {
+		const { data } = await axios.get(endpointFullString);
+		if (typeof data !== 'object') {
+			throw new Error('Response was not an object. Response acquired from the server: ' + data);
+		} else if (data.error !== null) {
+			throw new Error(data.error);
+		} else {
+			return data.data;
+		}
+	} catch (error) {
+		console.log(error);
+		return null;
+	}
+}
+
+function createApiRequestUrl(endpoint: string, queryParams: { [key: string]: any }): string {
```

Comment:
In this particular situation it's okay to create URLs that have `?` or `&` at the end because no one is going to look at them, but if you needed a similar function for creating user-facing URLs you'd have to make sure it looks good too: no question mark and & at the end of the URL.

Can you change it to satisfy those requirements? :) Btw, if you want to write it a bit more functional, you can use `Object.keys()` to get an array of keys which you can then, perhaps, map over.